### PR TITLE
Replace tokens for matched grps in regex

### PR DESCRIPTION
**Pull request by harshshah2-crest**
- Added user warnings while error in ingestions.
- Updated code for grp regex replacement.
- Added warning if unidentified input_type is obtained from conf file.

### CODE REVIEW

File: pytest_splunk_addon/standard_lib/event_ingestors/hec_event_ingestor.py [hunk 1]
```diff
@@ -111,4 +112,5 @@ def __ingest(self, data):
         
         except Exception as e:
             LOGGER.error(e)
+            warnings.warn(UserWarning("Error occcured while ingestion: {}".format(e)))
             os._exit(0)
```
Comment (jayjoshi-crest) [inline pytest_splunk_addon/standard_lib/event_ingestors/hec_event_ingestor.py:116]: warnings don't work if we os.exit(). Please raise an Exception instead.

File: pytest_splunk_addon/standard_lib/event_ingestors/hec_raw_ingestor.py [hunk 2]
```diff
@@ -90,4 +91,5 @@ def __ingest(self, event, params):
 
         except Exception as e:
             LOGGER.error(e)
+            warnings.warn(UserWarning("Error occcured while ingestion: {}".format(e)))
             os._exit(0)
```
Comment (jayjoshi-crest) [inline pytest_splunk_addon/standard_lib/event_ingestors/hec_raw_ingestor.py:95]: warnings don't work if we os.exit(). Please raise an Exception instead.

File: pytest_splunk_addon/standard_lib/sample_generation/sample_event.py [hunk 3]
```diff
@@ -172,10 +172,11 @@ def replace_token(self, token, token_values):
         # TODO: How to handle dependent Values with list of token_values
         if isinstance(token_values, list):
             sample_tokens = re.finditer(token, self.event)
-                
-            for token_index, token_value in enumerate(token_values):
+            for _, token_value in enumerate(token_values):
+                match_object = next(sample_tokens)
+                match_str = match_object.group(0) if len(match_object.groups()) == 0 else match_object.group(1)
```
Comment (jayjoshi-crest) [inline pytest_splunk_addon/standard_lib/sample_generation/sample_event.py:177]: Accurate.!

> If one or more capture groups are present the replacement will be performed on group 1.

Comment (harshshah2-crest) [reply pytest_splunk_addon/standard_lib/sample_generation/sample_event.py:177]: Yes exactly.

**Pull request merged**